apps/deploy-web/src/components/api-keys/CreateApiKeyModal.tsx

```diff
@@ -0,0 +1,132 @@
+import { useRef } from "react";
+import { useForm } from "react-hook-form";
+import { ActionButton, Button, Form, FormField, FormInput, Input, Popup, Snackbar } from "@akashnetwork/ui/components";
+import { copyTextToClipboard } from "@akashnetwork/ui/utils";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useSnackbar } from "notistack";
+import { z } from "zod";
+
+import { useCreateApiKey } from "@src/queries";
+import { analyticsService } from "@src/services/analytics/analytics.service";
+
+interface Props {
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+const formSchema = z.object({
+  name: z
+    .string()
+    .min(1, {
+      message: "Name is required."
+    })
+    .max(40, {
+      message: "Name must be less than 40 characters."
+    })
+});
+
+export const CreateApiKeyModal = ({ isOpen, onClose }: Props) => {
+  const { enqueueSnackbar } = useSnackbar();
+  const { mutate: createApiKey, data: createdApiKey, isLoading } = useCreateApiKey();
+  const formRef = useRef<HTMLFormElement | null>(null);
+  const form = useForm<z.infer<typeof formSchema>>({
+    defaultValues: {
+      name: ""
+    },
+    resolver: zodResolver(formSchema)
+  });
+  const {
+    handleSubmit,
+    control,
+    formState: { errors }
+  } = form;
+  const isCreatingNewKey = !createdApiKey;
+  const apiKey = createdApiKey?.apiKey || "";
+  const actions: ActionButton[] = isCreatingNewKey
+    ? [
+        {
+          label: "Cancel",
+          color: "primary",
+          variant: "secondary",
+          side: "left",
+          onClick: onClose
+        },
+        {
+          label: "Create Key",
+          color: "secondary",
+          variant: "default",
+          side: "right",
+          disabled: !!errors.name || isLoading,
+          isLoading,
+          onClick: event => {
+            event.preventDefault();
+            formRef.current?.dispatchEvent(new Event("submit", { cancelable: true, bubbles: true }));
+          }
+        }
+      ]
+    : [
+        {
+          label: "Done",
+          color: "primary",
+          variant: "secondary",
+          side: "right",
+          onClick: onClose
+        }
+      ];
+  const onCopyClick = () => {
+    copyTextToClipboard(apiKey);
+    enqueueSnackbar(<Snackbar title="Copied to clipboard!" iconVariant="success" />, { variant: "success", autoHideDuration: 1500 });
+  };
+
+  const onSubmit = async ({ name }) => {
```

Comment:
non-blocking. it would be nice to have this kind of functions reflect what they're doing. e.g. `createApiKeyTracked` or whatever that reflects the action.
Why: coz you might be looking at the invocation and have no idea what happens in the function, meaning you need to go and read the code.

e.g. compare the following
```tsx
          <form onSubmit={handleSubmit(onSubmit)} ref={formRef}> // no idea what happens on submit, go read the implementation
```

```tsx
          <form onSubmit={handleSubmit(createApiKeyTracked)} ref={formRef}> // ok, it would create the key and track smth. perhaps this is as far as I want to know
```

This is a general suggestion on how to approach these callback namings

Reply:
I don't mind giving it a shot